apps/src/applab/designElements/PropertyRow.jsx

```diff
@@ -9,17 +9,25 @@ var PropertyRow = React.createClass({
     handleChange: React.PropTypes.func
   },
 
+  getInitialState: function () {
+    return {
+      value: this.props.initialValue
+    };
+  },
+
   handleChangeInternal: function(event) {
     var value = event.target.value;
     this.props.handleChange(value);
+    this.setState({value: value});
   },
+
   render: function() {
     return (
       <tr>
         <td>{this.props.desc}</td>
         <td>
           <input
-            defaultValue={this.props.initialValue}
+            value={this.state.value}
```

Comment:
The "trick" was actually in a previous commit, where we gave the parent element a key. I don't remember exactly why in this PR I then also changed this to use value instead of default value. I think because using the defaultValue equivalent on my checkbox was giving me problems, and I was worried I would run into those here too.